File: TransitionDiagram/StratumTabStripItem.vb
﻿'************************************************************************************
' ST-Sim: A .NET class library for developing state-and-transition simulation models.
'
' Copyright © 2009-2015 ApexRMS.
'
'************************************************************************************

Imports SyncroSim.Core

Class StratumTabStripItem
    Inherits TransitionDiagramTabStripItem

    Private m_StratumId As Nullable(Of Integer)

    Public Sub New(ByVal stratumName As String, ByVal stratumId As Nullable(Of Integer))

        MyBase.New(stratumName)
        Me.m_StratumId = stratumId

    End Sub

    Public ReadOnly Property StratumId As Nullable(Of Integer)
        Get
            Return Me.m_StratumId
        End Get
    End Property

End Class

File: src/Diagram/StratumTabStripItem.cs
// stsim: A SyncroSim Package for developing state-and-transition simulation models using ST-Sim.
// Copyright © 2007-2024 Apex Resource Management Solutions Ltd. (ApexRMS). All rights reserved.

namespace SyncroSim.STSim
{
    internal class StratumTabStripItem : TransitionDiagramTabStripItem
    {
        private int? m_StratumId;

        public StratumTabStripItem(string stratumName, int? stratumId) : base(stratumName)
        {
            this.m_StratumId = stratumId;
        }

        public int? StratumId
        {
            get
            {
                return this.m_StratumId;
            }
        }
    }
}
